_user_guides/dinghy.md

```diff
@@ -290,3 +290,123 @@ The spinnaker `pipeline` stage allows you to trigger other pipelines. However, t
+
+### Newlines
+For ease of readablilty, you can split a single call to `module` across multiple lines. For example, the following two `dinghyfile`s are both valid & produce identical pipelines in spinnaker:
+```{% raw %}
+{
+  "application": "yourspinnakerapplicationname",
+  "pipelines": [
+    {
+      "application": "yourspinnakerapplicationname",
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {{ module "wait.stage.module" "name" "wait-for-cache-warm-up" "waitTime" 42 }}
+      ]
+    }
+  ]
+}
+{% endraw %}```
+
+```{% raw %}
+{
+  "application": "yourspinnakerapplicationname",
+  "pipelines": [
+    {
+      "application": "yourspinnakerapplicationname",
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {{
+           module "wait.stage.module"
+           "name" "wait-for-cache-warm-up"
+           "waitTime" 42
+        }}
+      ]
+    }
+  ]
+}
+{% endraw %}```
+
+### Top-level variables in dinghyfiles
+When passing in variables to modules, you have the option of defining variables at the top-level `dinghyfile` like so:
+```{% raw %}
+{
+  "application": "yourspinnakerapplicationname",
+  "globals": {
+      "waitTime": "42",
+      "name": "default-name"
+  },
+  "pipelines": [
+    {
+      "application": "yourspinnakerapplicationname",
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {{ module "wait.stage.module" }}
+      ]
+    }
+  ]
+}
+{% endraw %}```
+In the above example, the varialbes `waitTime` and `name` (used inside `wait.stage.module`) are defined at the top level, and not explicitly defined when the call to `wait.stage.module` is made.
+
+Note that top-level variables are overwritten by variables in the call to module if both are present. For instance, in the below example, the `waitTime` after the `dinghyfile` is rendered would be `43`:
+```{% raw %}
+{
+  "application": "yourspinnakerapplicationname",
+  "globals": {
+      "waitTime": "42",
+      "name": "default-name"
+  },
+  "pipelines": [
+    {
+      "application": "yourspinnakerapplicationname",
+      "name": "Made By Armory Pipeline Templates",
+      "stages": [
+        {{ module "wait.stage.module" "waitTime": "43" }}
+      ]
+    }
+  ]
+}
+{% endraw %}```
+
+### Nested variables
+Another neat little trick with variables is support for nested variables. Consider the following variable usage in a module:
+```{% raw %}
+"waitTime": {{ var "name" ?: "some-name" }}
+{% endraw %}```
+Here , we say if the variabled `"name"` was passed in, or is a top-level variable in the `dinghyfile`, then use that value, else _default to_ `some-name`.
+
+With nested variables, instead of using a hardcoded default value, the default can from another variable. eg:
```

Comment:
I think this is missing a word, maybe "the default can come from another variable"? (missing "come"?) . Or maybe "the default can be another variable" (change "from" to "be")